### PR TITLE
Fix #11989: Remove dependency on find

### LINKED ISSUE #11989: find: command not found on openjdk-oraclelinux8
I was setting up the scala 3.0.0-RC2 docker build at https://github.com/hseeberger/scala-sbt but the oraclelinux jdk build fails 

## Compiler version

3.0.0-RC2

## Minimized code

docker image `openjdk:11.0.10-jdk-oraclelinux8`

```Scala
@main def main = println(util.Properties.versionMsg)
```

## Output

```
scala test.scala && rm test.scala
#6 4.980 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.982 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.984 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.987 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.989 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.992 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.994 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.997 /usr/share/scala/bin/common: line 136: find: command not found
#6 4.999 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.002 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.004 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.007 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.010 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.032 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.034 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.037 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.041 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.044 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.047 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.049 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.052 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.056 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.058 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.061 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.063 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.065 /usr/share/scala/bin/common: line 136: find: command not found
#6 5.352 Error: Could not find or load main class dotty.tools.scripting.Main
#6 5.352 Caused by: java.lang.ClassNotFoundException: dotty.tools.scripting.Main
```

## Expectation

The scala command works as expected like on debian / graalvm docker images or with Scala 2.x

## Reproducer in docker build

1. check out master on https://github.com/hseeberger/scala-sbt
2. cd oracle
3. edit Dockerfile to mach line below
    `ENV SCALA_VERSION ${SCALA_VERSION:-3.0.0-RC2}`
4. remove section that installs findutils
5. docker build .

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #11989: Remove dependency on find

### CODE REVIEW

**Comment by liufengyun** [comment]
@michelou Do you have suggestions for removing the dependency on `find`?

**Comment by smarter** [comment]
We could emit jars without the version number in the filename so we can refer to them with a fixed identifier.

**Comment by michelou** [comment]
> @michelou Do you have suggestions for removing the dependency on `find`?

@liufengyun Let me start with two observations : 
1. I'm not aware of any rule/constraint about the organisation of the jar files inside the directory `<scala_home>/lib/`.
  **NB.** Using `ls` instead of `find` means that we assume a flat organisation of the jar files.
2. As a hot reaction I would be inclined to say that a standard Unix distribution should include a command `find`.

So the point is to choose if we want to adapt the bash scripts to yet another user environment **or** we decide to introduce a requirement about the user environment.

Nevertheless I acknowledge the fact that people are working more and more with Docker images (which may provide very minimal sets of commands).

Side note : IMO it would worth looking at other JVM based tools (Java, Kotlin, etc.) regarding the jar file organisation.

**Reminder** :  `ls` only applies to the current working directory, while `find` applies to all files and subdirectories starting from the current working directory.

**Comment by liufengyun** [comment]
Thanks @michelou.

> I'm not aware of any rule/constraint about the organisation of the jar files inside the directory <scala_home>/lib/

Yes, the directory is flat, which is a safe assumption in our case.

`ls` seems to work fine on Mac, but Linux is not happy.

**Comment by griggt** [comment]
Another option may be something like this, which eliminates both the subshell and the dependency on an external command like `find`/ `ls` / etc.

```shell
find_lib () {
  for lib in $PROG_HOME/lib/$1 ; do
    if [[ -e $lib ]]; then
      if [ -n "$CYGPATHCMD" ]; then
        "$CYGPATHCMD" -am $lib
      elif [[ $mingw ||  $msys ]]; then
        echo $lib | sed 's|/|\\\\|g'
      else
        echo $lib
      fi
      return
    fi
  done
}
```

**Comment by michelou** [comment]
> Another option may be something like this, which eliminates both the subshell and the dependency on an external command like `find`/ `ls` / etc.
> 
> ```shell
> find_lib () {
>   for lib in "$PROG_HOME"/lib/$1 ; do
>     if [[ -e "$lib" ]]; then
>       if [ -n "$CYGPATHCMD" ]; then
>         "$CYGPATHCMD" -am "$lib"
>       elif [[ $mingw ||  $msys ]]; then
>         echo "$lib" | sed 's|/|\\\\|g'
>       else
>         echo "$lib"
>       fi
>       return
>     fi
>   done
> }
> ```

On line 137, `[[ -e "$lib" ]]` may also include subdirectories. Why do you not use option `-f` (regular files only) here ?

File: dist/bin/common [hunk 1]
```diff
@@ -133,14 +133,18 @@ fi
...
-  local lib=$(find $PROG_HOME/lib/ -name "$1")
-  if [ -n "$CYGPATHCMD" ]; then
-    "$CYGPATHCMD" -am $lib
-  elif [[ $mingw ||  $msys ]]; then
-    echo $lib | sed 's|/|\\\\|g'
-  else
-    echo $lib
-  fi
+  for lib in "$PROG_HOME"/lib/$1 ; do
+    if [[ -e "$lib" ]]; then
```
Comment (griggt) [inline dist/bin/common:137]:
```suggestion
    if [[ -f "$lib" ]]; then
```

As pointed out by @michelou (thanks!), we can make this more robust by using `-f` here.

**Review by griggt — APPROVED** [review]
Otherwise LGTM.

**Pull request merged**